hazelcast/src/main/java/com/hazelcast/scheduledexecutor/IScheduledExecutorService.java

```diff
@@ -0,0 +1,395 @@
+ * To emphasize this difference the method is called <code>scheduleWithRepetition</code>
+ * instead of <code>scheduleAtFixedRate</code>
+ * <br/><br/>
+ * The other difference is this service does not offer an equivalent of
+ * {@link java.util.concurrent.ScheduledExecutorService#scheduleWithFixedDelay(Runnable, long, long, TimeUnit)}
+ */
+@Beta
+public interface IScheduledExecutorService extends DistributedObject {
+
+    /**
+     * Creates and executes a one-shot action that becomes enabled
+     * after the given delay.
+     *
+     * @param command the task to execute
+     * @param delay the time from now to delay execution
+     * @param unit the time unit of the delay parameter
+     * @return a ScheduledFuture representing pending completion of
+     *         the task and whose {@code get()} method will return
+     *         {@code null} upon completion
+     * @throws RejectedExecutionException if the task cannot be
+     *         scheduled for execution
+     * @throws NullPointerException if command is null
+     */
+    IScheduledFuture<?> schedule(Runnable command, long delay, TimeUnit unit);
+
+    /**
+     * Creates and executes a one-shot action that becomes enabled
+     * after the given delay.
+     *
+     * @param command the task to execute
+     * @param delay the time from now to delay execution
+     * @param unit the time unit of the delay parameter
+     * @return a ScheduledFuture representing pending completion of
+     *         the task and whose {@code get()} method will return
+     *         {@code null} upon completion
+     * @throws RejectedExecutionException if the task cannot be
+     *         scheduled for execution
+     * @throws NullPointerException if command is null
+     */
+    <V> IScheduledFuture<V> schedule(Callable<V> command, long delay, TimeUnit unit);
+
+    /**
+     * Creates and executes a periodic action that becomes enabled first
+     * after the given initial delay, and subsequently with the given
+     * period. Executions will commence after
+     * {@code initialDelay} then {@code initialDelay+period}, then
+     * {@code initialDelay + 2 * period}, and so on.
+     * If any execution of this task
+     * takes longer than its period, then subsequent execution will be skipped.
+     *
+     * @param command the task to execute
+     * @param initialDelay the time to delay first execution
+     * @param period the period between successive executions
+     * @param unit the time unit of the initialDelay and period parameters
+     * @return a ScheduledFuture representing pending completion of
+     *         the task, and whose {@code get()} method will throw an
+     *         exception upon cancellation
+     * @throws RejectedExecutionException if the task cannot be
+     *         scheduled for execution
+     * @throws NullPointerException if command is null
+     */
+    IScheduledFuture<?> scheduleWithRepetition(Runnable command, long initialDelay,
```

Comment:
There is class Javadoc for the difference

> The difference is that this service will skip a scheduled execution if another thread is still running the same task, instead of postponing its execution.
> To emphasize this difference the method is called <code>scheduleWithRepetition</code>